main.js

```diff
@@ -0,0 +1,446 @@
+const searchInput = document.querySelector('.search-input');
+const searchButton = document.querySelector('.search-button');
+const weatherInfo = document.querySelector('.weather-info');
+const errorMessage = document.querySelector('.error-message');
+const weatherIcon = document.querySelector('.weather-icon');
+const temperatureElement = document.querySelector('.temperature');
+const descriptionElement = document.querySelector('.description');
+const locationElement = document.querySelector('.location');
+const loadingSpinner = document.querySelector('.loading-spinner');
+const apiKey = '82005d27a116c2880c8f0fcb866998a0';
+
+const weatherIcons = {
+    Clear: '01d.png',
+    Clouds: '04d.png',
+    Rain: '09d.png',
+    Drizzle: '10d.png',
+    Thunderstorm: '11d.png',
+    Snow: '13d.png',
+    Mist: '50d.png',
+    Smoke: '10n.png',
+    Haze: '02d.png',
+    Fog: '09n.png',
+    Sand: '02n.png',
+    Ash: '03d.png',
+    Squall: 'unknown.png',
+};
+
+
+function WeatherDetails() {
```

Comment:
1. function name should start with smallLetter
2. name should be descriptive, WeatherDetails not making any sense